lib/internal/modules/esm/fetch_module.js

```diff
@@ -99,118 +96,75 @@ function fetchWithRedirects(parsed) {
     return existing;
   }
   const handler = parsed.protocol === 'http:' ? HTTPGet : HTTPSGet;
-  const result = new Promise((fulfill, reject) => {
+  const result = (async () => {
     const req = handler(parsed, {
-      headers: {
-        Accept: '*/*'
-      }
-    })
-    .on('error', reject)
-    .on('response', (res) => {
-      function dispose() {
-        req.destroy();
-        res.destroy();
-      }
+      headers: { Accept: '*/*' },
+    });
+    const { 0: res } = await once(req, 'response');
```

Comment:
please add note that this handles "error"

Reply:
I had a question about that by the way - currently even if the request fails we cache it so multiple `import()` calls will fail. I thought it might make sense to remove the module from the cache on things like 404s but I am not sure what browsers do. 

Anyway that change is probably (?) out of scope for this PR which just refactors the code to async/await and compose mostly.

Reply:
yup, cache 404s same as ENOENT for `file:`, there is a potential desire to alter this in the future but right now cache em.

Reply:
adding a note about error in case of network / timeout would be good just so people don't assume getting to after await is guaranteed eventually